packages/backend/src/graphql/mutations/delete-step.ts

```diff
@@ -19,20 +20,22 @@ const deleteStep = async (
     })
     .throwIfNotFound()
 
-  await step.$relatedQuery('executionSteps').delete()
-  await step.$query().delete()
+  await Step.transaction(async (trx) => {
+    await step.$relatedQuery('executionSteps', trx).delete()
+    await step.$query(trx).delete()
 
-  const nextSteps = await step.flow
-    .$relatedQuery('steps')
-    .where('position', '>', step.position)
+    const nextSteps = await step.flow
+      .$relatedQuery('steps')
+      .where('position', '>', step.position)
 
-  const nextStepQueries = nextSteps.map(async (nextStep) => {
-    await nextStep.$query().patch({
-      position: nextStep.position - 1,
+    const nextStepQueries = nextSteps.map(async (nextStep) => {
+      await nextStep.$query().patch({
+        position: nextStep.position - 1,
+      })
     })
-  })
 
-  await Promise.all(nextStepQueries)
+    await Promise.all(nextStepQueries)
+  })
```

Comment:
good use of transaction here but can also include the select and patch queries for `nextSteps` in the transaction as well?

Reply:
may can wrap create-step also

Reply:
Wah good catch! Forgot to pass down `trx` to the other queries